nilearn/conftest.py

```diff
@@ -336,3 +329,112 @@ def img_atlas(shape_3d_default, affine_mni):
             "csf": 3,
         },
     }
+
+
+# ------------------------ SURFACE ------------------------#
+
+
+@pytest.fixture
+def mini_mesh():
+    """Small mesh for tests with 2 parts with different numbers of vertices."""
+    left_coords = np.asarray([[0.0, 0, 0], [1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    left_faces = np.asarray([[1, 0, 2], [0, 1, 3], [0, 3, 2], [1, 2, 3]])
+    right_coords = (
+        np.asarray([[0.0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 1, 0], [0, 0, 1]])
+        + 2.0
+    )
+    right_faces = np.asarray(
+        [
+            [0, 1, 4],
+            [0, 3, 1],
+            [1, 3, 2],
+            [1, 2, 4],
+            [2, 3, 4],
+            [0, 4, 3],
+        ]
+    )
+    return {
+        "left_hemisphere": InMemoryMesh(left_coords, left_faces),
+        "right_hemisphere": InMemoryMesh(right_coords, right_faces),
+    }
+
+
+@pytest.fixture
+def make_mini_img(mini_mesh):
+    """Small surface image for tests."""
+
+    def f(shape=()):
+        data = {}
+        for i, (key, val) in enumerate(mini_mesh.items()):
+            data_shape = tuple(shape) + (val.n_vertices,)
+            data_part = (
+                np.arange(np.prod(data_shape)).reshape(data_shape) + 1.0
+            ) * 10**i
+            data[key] = data_part
+        return SurfaceImage(mini_mesh, data)
+
+    return f
+
+
+@pytest.fixture
+def mini_mask(mini_img):
+    """Small surface mask."""
+    data = {k: (v > v.ravel()[0]) for k, v in mini_img.data.items()}
+    return SurfaceImage(mini_img.mesh, data)
+
+
+@pytest.fixture
+def mini_img(make_mini_img):
+    """Small surface image for tests."""
+    return make_mini_img()
+
+
+@pytest.fixture
+def flip():
+    """Reorder dictionary keys by putting the last key as the first."""
+
+    def f(parts):
+        if not parts:
+            return {}
+        keys = list(parts.keys())
+        keys = [keys[-1]] + keys[:-1]
+        return dict(zip(keys, parts.values()))
+
+    return f
+
+
+@pytest.fixture
+def flip_img(flip):
+    """Reorder dictionary keys by putting the last key as the first for mesh \
+       and data parts of SurfaceImage object."""
+
+    def f(img):
+        return SurfaceImage(flip(img.mesh), flip(img.data))
+
+    return f
+
+
+@pytest.fixture
+def assert_img_equal():
```

Comment:
```suggestion
def assert_surf_img_equal():
```

to avoid confusion with volume image

also check if it needs to be a structure